ocs_ci/utility/tnf_hypervisor.py

```
@@ -0,0 +1,937 @@
+        """
+        Generate dev-scripts config file and upload pull secret.
+        """
+        dev_scripts_dir = constants.TNF_DEV_SCRIPTS_DIR
+        cluster_name = self.dev_scripts_config.get("cluster_name", "tnf-cluster")
+        remote_ps_path = f"{dev_scripts_dir}/pull_secret.json"
+
+        with tempfile.NamedTemporaryFile(mode="w", suffix=".json", delete=False) as f:
+            f.write(pull_secret_content)
+            local_ps_path = f.name
+
+        tmp_ps_path = "/tmp/pull_secret.json"
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/pull_secret.json"

(S108)